ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/command/AbstractParentCommand.java

```diff
@@ -24,13 +24,13 @@
 import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
-public abstract class AbstractParentCommand extends AbstractRatisCommand{
+public abstract class AbstractParentCommand implements Command {
   private final Map<String, Command> subs;
 
   public AbstractParentCommand(Context context, List<Function<Context, AbstractRatisCommand>> subCommandConstructors) {
```

Comment:
Let's change the `Function` to `Function<Context, Command>`.  Then, the new `LocalCommand` can also use it.